modules/standard/CTypes.chpl

```diff
@@ -355,6 +355,11 @@ module CTypes {
         return true;
       }
       return pointeeCastStrictAliasingAllowed(from.eltType, to.eltType);
+    } else if (from == c_string) {
+      // a c_string can be interpreted as a pointer to c_char for these purposes
+      return pointeeCastStrictAliasingAllowed(c_char, to.eltType);
+    } else if (to == c_string) {
+      return pointeeCastStrictAliasingAllowed(from.eltType, c_char);
```

Comment:
Here and above, it doesn't seem a sure thing that from has an `.eltType` so you probably should include `&& isAnyCPtr` in the conditionals besides the `from == c_string` or `to == c_string`. (Of course I could be missing something). If you do that, consider it approved by me.

(Also I think it's odd that `isAnyCPtr` seems to return false for `c_string` -- if that's not the case you would need to write these conditionals differently -- but anyway I think it's a preexisting issue.)

Reply:
Good catch, I do need to check that.

(I also was surprised by the fact that `isAnyCPtr` returns false for `c_string`. However I don't think it's worth changing since `c_string` is to be replaced by a `c_ptrConst(c_char)` relatively soon anyways.)